contracts/FragToken.sol

```diff
@@ -70,40 +75,56 @@ contract FRAGToken is ERC20, ERC20Permit, Ownable{
             "Invalid signature"
         );
 
-        // add to current locked amount
-        _locksAmount[msg.sender] = _locksAmount[msg.sender] + amount;
+        _lockInfo.sender = msg.sender;
```

Comment:
Using a temporary stack variable here is the way to go no?

Reply:
Of course....created a local temporary `memory` variable for the struct.